cardano-testnet/src/Test/Runtime.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Test.Runtime
+  ( NodeLoggingFormat(..)
+  , PaymentKeyPair(..)
+  , StakingKeyPair(..)
+  , TestnetRuntime(..)
+  , TestnetNode(..)
+  , PoolNode(..)
+  , PoolNodeKeys(..)
+  , Delegator(..)
+  , bftSprockets
+  , poolSprockets
+  , poolNodeToTestnetNode
+  , readNodeLoggingFormat
+  ) where
+
+import           Data.Either (Either (..))
+import           Data.Eq (Eq)
+import           Data.Function (($), (.))
+import           Data.Functor (fmap)
+import           Data.Int (Int)
+import           Data.Semigroup (Semigroup ((<>)))
+import           Data.String (String)
+import           Hedgehog.Extras.Stock.IO.Network.Sprocket (Sprocket (..))
+import           System.IO (FilePath)
+import           Text.Show (Show (..))
+
+import qualified System.IO as IO
+import qualified System.Process as IO
+
+data NodeLoggingFormat = NodeLoggingFormatAsJson | NodeLoggingFormatAsText deriving (Eq, Show)
+
+data TestnetRuntime = TestnetRuntime
+  { configurationFile :: FilePath
+  , shelleyGenesisFile :: FilePath
+  , testnetMagic :: Int
+  , bftNodes :: [TestnetNode]
+  , poolNodes :: [PoolNode]
+  , wallets :: [PaymentKeyPair]
+  , delegators :: [Delegator]
+  }
+
+data TestnetNode = TestnetNode
+  { nodeName :: String
+  , nodeSprocket :: Sprocket
+  , nodeStdinHandle :: IO.Handle
+  , nodeStdout :: FilePath
+  , nodeStderr :: FilePath
+  , nodeProcessHandle :: IO.ProcessHandle
+  }
+
+data PoolNode = PoolNode
+  { poolNodeName :: String
+  , poolNodeSprocket :: Sprocket
+  , poolNodeStdinHandle :: IO.Handle
+  , poolNodeStdout :: FilePath
+  , poolNodeStderr :: FilePath
+  , poolNodeProcessHandle :: IO.ProcessHandle
+  , poolNodeKeys :: PoolNodeKeys
+  }
+
+data PoolNodeKeys = PoolNodeKeys
+  { poolNodeKeysColdVkey :: FilePath
```

Comment:
It  would be nice to have the actual `cardano-api` types for the keys here but not for this PR (if this is actually possible). Why? Just to separate out as much of the IO as possible.